server/src/internalClusterTest/java/org/elasticsearch/snapshots/SystemIndicesSnapshotIT.java

```diff
@@ -608,6 +711,69 @@ public void testSystemIndexAliasesAreAlwaysRestored() {
 
     }
 
+    // TODO: This test is failing because the assertion `assertTrue(indexExists(systemDataStreamAlias));` trips.
+    @AwaitsFix(bugUrl = "assertTrue(indexExists(systemDataStreamAlias)); is broken")
```

Comment:
This test is failing because the alias isn't restored. I'm supposing this is a bug, and not expected behaviour for system datastreams?

Reply:
Data stream aliases are similar to index aliases, where they are metadata on the data stream itself. I would expect restoring a data stream (system or otherwise) to also restore the alias for that data stream.

I tested that with a non-system data stream, the alias is restored when the data stream is restored, so I would treat this as a bug.

Reply:
Thanks @dakrone , I found the bug, fixed it, and enabled the test.